db/kv.go

```diff
@@ -320,7 +320,8 @@ func (kv *kv) getVersion(name string, version api.SecretVersion) (*api.SecretVal
 // exists, value is saved as a new inactive version. Otherwise, value
 // is saved as the initial version of the secret and immediately set
 // active. On success, returns the secret version for the new value.
-func (kv *kv) put(name string, value []byte) (api.SecretVersion, error) {
+// If an explicit version is specified, the value is saved at that version.
+func (kv *kv) put(name string, value []byte, version *api.SecretVersion) (api.SecretVersion, error) {
```

Comment:
Let's not use a pointer for this; we can use 0 as a sentinel value since it's not a valid version otherwise. It'll be nicer (IMO) if we let 0 mean "assign me a number".

Also: Versions are currently uint32, and if you want to use them as timestamps, we should make that uint64 (see types/api). For most of the code this will be a relatively minor change, however:

Relatedly: Now that the user is allowed to provide a value, we will need to deal with JSON number encoding properly (the usual 53-bit/64-bit issue).